### PR TITLE
Cert auth support

### PR DESCRIPTION
**Pull request by aleklisi**
This PR adds docs to the main PR.

### CODE REVIEW

File: doc/authentication-backends/HTTP-authentication-module.md [hunk 1]
```diff
@@ -112,6 +112,15 @@ A body can be missing in the first data chunk read from a socket, leading to str
     * 200, `true` or `false` in body
     * anything else - will be treated as `false`
 
+### Method `get_certs`
+
+* **Description** Provides a list of certificates corresponding to a given user. Works like third party Certificate Authority service. The certificates must be given in `PEM` format.
```
Comment (goddammit) [inline doc/authentication-backends/HTTP-authentication-module.md]: **Description** Provides a list of certificates corresponding to a given user. Works like a third party Certificate Authority service. The certificates must be given in a `PEM` format.

File: doc/authentication-backends/HTTP-authentication-module.md [hunk 2]
```diff
@@ -112,6 +112,15 @@ A body can be missing in the first data chunk read from a socket, leading to str
...
+* **HTTP method** GET
+* **Type**: mandatory when verification with `ejabberd_auth_http` is used
+* **Return values**:
+  * `200`, when user exists, body consists of a list of certificates corresponding to user
```
Comment (goddammit) [inline doc/authentication-backends/HTTP-authentication-module.md]: * `200`, when a user exists, the body consists of a list of certificates corresponding to them

File: src/sasl/cyrsasl_external_verification.erl [hunk 3]
```diff
@@ -3,7 +3,11 @@
 %%% @author Denys Gonchar <denys.gonchar@erlang-solutions.com>
 %%% @doc cyrsasl external verification backend
 %%%
-%%% this module is added for demo & testing purposes only.
+%%% This is experimental prototype module for demo & testing purposes only.
```
Comment (goddammit) [inline src/sasl/cyrsasl_external_verification.erl]: %%% This is an experimental prototype module for demo & testing purposes only.

File: src/sasl/cyrsasl_external_verification.erl [hunk 4]
```diff
@@ -3,7 +3,11 @@
...
+%%% It allows to get a list of certificates from third party Certificate Authority service.
```
Comment (goddammit) [inline src/sasl/cyrsasl_external_verification.erl]: %%% It allows to get a list of certificates from a third party Certificate Authority service.

File: src/sasl/cyrsasl_external_verification.erl [hunk 5]
```diff
@@ -3,7 +3,11 @@
...
+%%% The certificates must be given in `PEM` format.
```
Comment (goddammit) [inline src/sasl/cyrsasl_external_verification.erl]: %%% The certificates must be given in a `PEM` format.

File: src/sasl/cyrsasl_external_verification.erl [hunk 6]
```diff
@@ -3,7 +3,11 @@
...
+%%% If user certificate matches any of the certificates form the list,
```
Comment (goddammit) [inline src/sasl/cyrsasl_external_verification.erl]: If the user certificate matches any of the certificates form the list,

File: src/sasl/cyrsasl_external_verification.erl [hunk 7]
```diff
@@ -3,7 +3,11 @@
 %%% @author Denys Gonchar <denys.gonchar@erlang-solutions.com>
 %%% @doc cyrsasl external verification backend
 %%%
-%%% this module is added for demo & testing purposes only.
+%%% This is an experimental prototype module for demo & testing purposes only.
+%%% It allows to get a list of certificates from a third party Certificate Authority service.
```
Comment (DenysGonchar) [inline src/sasl/cyrsasl_external_verification.erl]: it's completely unrelated to http auth. everything written here is wrong!

Comment (aleklisi) [reply src/sasl/cyrsasl_external_verification.erl]: so I just delete this line? or is the whole comment wrong?

Comment (DenysGonchar) [reply src/sasl/cyrsasl_external_verification.erl]: the whole statement

**Pull request closed without merging**